scripts/compress_images_test.py

```diff
@@ -0,0 +1,250 @@
+            args,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+    def test_run_no_compressible_images(self) -> None:
+        """Test run method when no images are compressible."""
+        temp_dir = tempfile.mkdtemp()
+        with open(self.txt_image_path, 'w', encoding='utf-8') as f:
+            f.write('This is not a valid image.')
+
+        compressor = compress_images.ImageCompressor(temp_dir)
+        result = compressor.run()
+        self.assertEqual(result, 0)
+
+    def test_run_with_compressible_images(self) -> None:
+        """Test run method with compressible images."""
+
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            self.output_dir
+        )
+        result = compressor.run()
+        self.assertEqual(result, 1)
+
+    def test_unsupported_file_extensions(self) -> None:
+        """Test handling of unsupported file extensions."""
+        with open(self.txt_image_path, 'w', encoding='utf-8') as f:
+            f.write('Not an image')
+
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compressible_images = compressor.find_compressible_images()
+        self.assertEqual(len(compressible_images), 1)
+
+    def test_compress_images_handles_compression_failure(self) -> None:
+        """Test handling of compression failure for an image."""
+        result_images: List[CompressedImageInfo] = [
+            {
+                'path': pathlib.Path(self.png_path),
+                'original_size': 1000,
+                'new_size': 500
+            }
+        ]
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        with (
+            mock.patch('subprocess.run') as mock_subprocess_run,
+            mock.patch('logging.error') as mock_log_error
+        ):
+            mock_subprocess_run.return_value = mock.Mock(
+                returncode=1,
+                stderr='Compression error message'
+            )
+            with mock.patch('PIL.Image.open'):
+                compressor.compress_images(result_images)
+
+            mock_log_error.assert_called_once_with(
+                '[ERROR]: %s occurred on file %s',
+                'Compression error message',
+                pathlib.Path(self.png_path)
+            )
+            args = [
+                'gm', 'convert',
+                str(self.png_path),
+                '-strip',
+                '-compress', 'Zip',
+                mock.ANY
+            ]
+            mock_subprocess_run.assert_called_once_with(
+                args,
+                capture_output=True,
+                text=True,
+                check=True
+            )
+
+    def test_main_function(self) -> None:
+        """Test the main function execution."""
+        mock_run = mock.MagicMock(return_value=0)
+        mock_compressor = mock.MagicMock()
```

Comment:
here and elsewhere, please use `spec` to specify the expected attributes (or lack thereof)